pkg/controllers/node/termination/terminator/terminator.go

```diff
@@ -68,23 +79,27 @@ func (t *Terminator) Cordon(ctx context.Context, node *v1.Node) error {
 // https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
 func (t *Terminator) Drain(ctx context.Context, node *v1.Node) error {
 	// Get evictable pods
-	pods, err := t.getPods(ctx, node)
-	if err != nil {
-		return fmt.Errorf("listing pods for node, %w", err)
+	pods := &v1.PodList{}
+	if err := t.kubeClient.List(ctx, pods, client.MatchingFields{"spec.nodeName": node.Name}); err != nil {
+		return fmt.Errorf("listing pods on node, %w", err)
 	}
-	var podsToEvict []*v1.Pod
+
+	_, isMachine := node.Labels[v1alpha5.ProvisionerNameLabelKey]
 	// Skip node due to pods that are not able to be evicted
-	for _, p := range pods {
-		// Ignore if unschedulable is tolerated, since they will reschedule
-		if podutil.ToleratesUnschedulableTaint(p) {
-			continue
+	podsToEvict := lo.FilterMap(pods.Items, func(po v1.Pod, _ int) (*v1.Pod, bool) {
```

Comment:
Ah, I see that you are filtering out here but I wonder if this is best delegated to a helper function

Reply:
I think it's easier to see the filters that we apply if we do it in-line.

Reply:
Decided to punt on this and open a chore issue to look into this later